enumerables.rb

```diff
@@ -0,0 +1,137 @@
+# rubocop: disable Metrics/ModuleLength, Style/CaseEquality, Style/IfInsideElse
+
+module Enumerable
+  def my_each(var = nil)
+    return to_enum unless block_given?
+
+    n = if var.nil?
+          0
+        else
+          var
+        end
+    while n <= size - 1
+      yield(to_a[n])
+      n += 1
+    end
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    n = 0
+    while n <= size - 1
+      yield(to_a[n], n)
+      n += 1
+    end
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    result = []
+    my_each do |n|
+      result.push(n) if yield(n)
+    end
+    result
+  end
+
+  def my_all?(var = nil)
+    result = true
+    my_each do |n|
+      if block_given?
+        result = false unless yield(n)
+      elsif var.nil?
+        result = false unless n
+      else
+        result = false unless var === n
+      end
+    end
+    result
+  end
+
+  def my_any?(var = nil)
+    result = false
+    my_each do |n|
+      if block_given?
+        return result = true if yield(n)
+      elsif var.nil?
+        return result = true if n
+      else
+        return result = true if var === n
+      end
+    end
+    result
+  end
+
+  def my_none?(var = nil)
+    result = true
+    my_each do |n|
+      if block_given?
+        return result = false if yield(n)
+      elsif var.nil?
+        return result = false if n
+      else
+        return result = false if var === n
+      end
+    end
+    result
+  end
+
+  def my_count(var = nil)
+    result = []
+    if block_given?
+      my_each do |n|
+        result.push(n) if yield(n)
+      end
+    elsif var.nil?
+      return length
+    else
+      my_each do |n|
+        result.push(n) if n == var
+      end
+    end
+    result.length
+  end
+
+  def my_map(var = nil)
+    result = []
+    return to_enum unless block_given? && var.nil?
+
+    if var
+      my_each do |n|
+        result.push(var.call(n))
+      end
+    else
+      my_each do |n|
+        result.push(yield(n))
+      end
+    end
+    result
+  end
```

Comment:
[OPTIONAL]

#### my_map 

- [ ] in the specifications of the project in the [Odin project](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks) # 12, you were asked to modify your my_map method to take either a proc or a block. And if both a proc and a block are given, only execute the proc. The real map doesn't have this feature, but you were asked to have it so I strongly recommend working on it:

Given 
```ruby
my_proc = Proc.new {|num| num > 10 }
```
Result
```ruby
puts [18, 22, 5, 6] .my_map(my_proc) {|num| num < 10 }  => true true false false
```

Reply:
I have just implemented it, thank you for noticing! :smile: